### PR TITLE
Adding log message containing actual number of compares for Span API tests

### PR DESCRIPTION
**Pull request by ahsonkhan**
This will help us get more info about intermittent failures like:
https://github.com/dotnet/corefx/issues/23430

cc @AtsushiKan, @shiftylogic

### CODE REVIEW

File: src/System.Memory/tests/ReadOnlySpan/IndexOf.T.cs [hunk 1]
```diff
@@ -80,7 +80,8 @@ public static void OnNoMatchMakeSureEveryElementIsCompared()
                 foreach (TInt elem in a)
                 {
                     int numCompares = log.CountCompares(elem.Value, 9999);
-                    Assert.True(numCompares == 1);
+                    string message = string.Format("Expected number of compares is 1. Actual number of compares is {0}. Element value used to compare is {1}.", numCompares, elem.Value);
+                    Assert.True(numCompares == 1, message);
```
Comment (stephentoub) [inline src/System.Memory/tests/ReadOnlySpan/IndexOf.T.cs]: Nit: could be simplified to something like:
```C#
Assert.True(numCompares ==1, $"Expected {numCompares} == 1 for element {elem.Value}");
```

File: src/System.Memory/tests/ReadOnlySpan/SequenceEqual.T.cs [hunk 2]
```diff
@@ -64,7 +64,8 @@ public static void OnSequenceEqualOfEqualSpansMakeSureEveryElementIsCompared()
                 foreach (TInt elem in first)
                 {
                     int numCompares = log.CountCompares(elem.Value, elem.Value);
-                    Assert.True(numCompares == 1);
+                    string message = string.Format("Expected number of compares is 1. Actual number of compares is {0}. Element value used to compare is {1}.", numCompares, elem.Value);
+                    Assert.True(numCompares == 1, message);
```
Comment (stephentoub) [inline src/System.Memory/tests/ReadOnlySpan/SequenceEqual.T.cs]: Same nit

**Comment by joshfree** [comment]
👍

**Comment by ahsonkhan** [comment]
@dotnet-bot test Linux x64 Release Build

**Pull request merged**